lib/src/connection.rs

```diff
@@ -53,9 +64,47 @@ impl Connection {
         user: &str,
         password: &str,
     ) -> Result<Connection> {
+        let root_cert_store = Self::build_cert_store();
+        let stream = Self::build_stream(stream, host, root_cert_store).await?;
+
+        Self::init(user, password, stream).await
+    }
+
+    async fn new_tls_with_certificate<T: AsRef<str>>(
+        stream: TcpStream,
+        host: &Host<T>,
+        user: &str,
+        password: &str,
+        certificate: &ClientCertificate,
+    ) -> Result<Connection> {
+        let mut root_cert_store = Self::build_cert_store();
+
+        let cert_file = File::open(certificate.cert_file.as_str())?;
+        let mut reader = BufReader::new(cert_file);
+        let certs = rustls_pemfile::certs(&mut reader);
+        for certificate in certs {
+            if let Ok(cert) = certificate {
+                root_cert_store.add(cert).unwrap();
+            }
+        }
+
+        let stream = Self::build_stream(stream, host, root_cert_store).await?;
+        Self::init(user, password, stream).await
+    }
+
+    fn build_cert_store() -> RootCertStore {
         let mut root_cert_store = RootCertStore::empty();
-        root_cert_store.extend(webpki_roots::TLS_SERVER_ROOTS.iter().map(ToOwned::to_owned));
+        if let Ok(certs) = rustls_native_certs::load_native_certs() {
+            for cert in certs {
+                root_cert_store.add(cert).unwrap();
+            }
+        } else {
+            warn!("Failed to load native certificates!");
+        }
+        root_cert_store
+    }
```

Comment:
Similar here, we can use a less fallible API. In addition, I'd like to include the error in the warning log. Since the calling methods can already fail on an io error, this could also be returning the error directly, which would mean that an encrypted connection could not be used when the native certs could not be loaded, as opposed to it continuing with an empty cert store, which I think would be acceptable.

```suggestion
    fn build_cert_store() -> RootCertStore {
        let mut root_cert_store = RootCertStore::empty();
        match rustls_native_certs::load_native_certs() {
            Ok(certs) => {
                root_cert_store.add_parsable_certificates(certs);
            }
            Err(e) => {
                warn!("Failed to load native certificates: {e}");
            }
        }
        root_cert_store
    }
```